### PR TITLE
Fixes #55884- Added else for invalid fill_method

### LINKED ISSUE #55884: `merge_ordered()` doesn't validate `fill_method`
[The docs](https://pandas.pydata.org/docs/dev/reference/api/pandas.merge_ordered.html) say `fill_method` is the "Interpolation method for data", but I just tried `pd.merge_ordered(..., fill_method='linear')` and it didn't interpolate.

Looking closer at the docs, `fill_method` can only be either `'ffill'` or `None`, but apparently that's not actually checked; for example, `fill_method='carrot'` also works.

> **fill_method** : _{'ffill', None}, default None_
> Interpolation method for data.

Looks like this code is responsible:
https://github.com/pandas-dev/pandas/blob/a167f13876870067ba610b808919f6b85b6b761e/pandas/core/reshape/merge.py#L1859-L1868

It should probably be changed to something like:
```py
if self.fill_method == "ffill": 
    if left_indexer is None:
    ...
elif self.fill_method is None: 
    left_join_indexer = left_indexer
    right_join_indexer = right_indexer
else:
    raise ValueError(f"fill_method must be one of ['ffill']. Got '{self.fill_method}' instead.")
```

(ValueError text based on equivalent `df.interpolate` error)

### PR DESCRIPTION
**Pull request by ziadk**
- [x] Closes #55884 
- [x] Tests added and passed
- [x] All code checks passed

### CODE REVIEW

**Comment by ziadk** [comment]
@lithomas1 could you please review this ^^? Thanks a lot !

File: pandas/tests/reshape/merge/test_merge_ordered.py [hunk 1]
```diff
@@ -209,3 +211,24 @@ def test_elements_not_in_by_but_in_df(self):
         msg = r"\{'h'\} not found in left columns"
         with pytest.raises(KeyError, match=msg):
             merge_ordered(left, right, on="E", left_by=["G", "h"])
+
+    def test_ffill_5584(self):
+        # GH 5584
```
Comment (mroeschke) [inline pandas/tests/reshape/merge/test_merge_ordered.py:216]:
```suggestion
        # GH 55884
```

File: pandas/tests/reshape/merge/test_merge_ordered.py [hunk 2]
```diff
@@ -209,3 +211,24 @@ def test_elements_not_in_by_but_in_df(self):
         msg = r"\{'h'\} not found in left columns"
         with pytest.raises(KeyError, match=msg):
             merge_ordered(left, right, on="E", left_by=["G", "h"])
+
+    def test_ffill_5584(self):
```
Comment (mroeschke) [inline pandas/tests/reshape/merge/test_merge_ordered.py:215]:
```suggestion
    def test_ffill_validate_fill_method(self):
```

File: pandas/tests/reshape/merge/test_merge_ordered.py [hunk 3]
```diff
@@ -209,3 +211,24 @@ def test_elements_not_in_by_but_in_df(self):
...
+        # GH 5584
+        df1 = DataFrame({"key": ["a", "c", "e"], "lvalue": [1, 2, 3]})
+        df2 = DataFrame({"key": ["b", "c", "d", "f"], "rvalue": [4, 5, 6, 7]})
+
+        for valid_method in ["ffill", None]:
```
Comment (mroeschke) [inline pandas/tests/reshape/merge/test_merge_ordered.py:220]: I don't think you need to test these cases

File: pandas/tests/reshape/merge/test_merge_ordered.py [hunk 4]
```diff
@@ -209,3 +211,24 @@ def test_elements_not_in_by_but_in_df(self):
...
+            try:
+                merge_ordered(df1, df2, on="key", fill_method=valid_method)
+            except Exception:
+                pytest.fail(f"Unexpected error with valid fill_method: {valid_method}")
+
+        for invalid_method in ["linear", "carrot"]:
```
Comment (mroeschke) [inline pandas/tests/reshape/merge/test_merge_ordered.py:226]: Can you use `pytest.mark.parametrize` here?

**Comment by ziadk** [comment]
@mroeschke All suggestions were implemented. Thanks a lot for the guidance. This was very instructive !

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @ziadk

**Pull request merged**